### PR TITLE
764 ci docs

### PR DESCRIPTION
**Pull request by dcolligan**
depends on #763

### CODE REVIEW

File: docs/development.rst [hunk 1]
```diff
@@ -323,3 +328,104 @@ Where ``desiredVersion`` is the version that will be written to the
...
+The difficulty of proceeding the opposite way (rebasing, then squashing) is
+only compounded because of the unintuitiveness of the N merge conflicts.
+When presented with a merge conflict, your likely intuition is to put the
+file in the state that you think it ought to be in, namely the condition it was
+in after the Nth commit.  However, if that state was different than the
+state that git thinks it should be in -- namely, the state of the file at
+commit X where X<N -- then you have only created the potential for more
+merge conflicts.  When the next intermediate commit, Y (where X<Y<N) is
+applied, it too will create a merge conflict.  And so on.
+
```
Comment (jeromekelleher) [inline docs/development.rst]: This is very good advice, which I didn't know about. However, I don't know how you'd actually do it -- can you put in a command example of the commands required to squash a topic branch, and then rebase?

**Comment by jeromekelleher** [comment]
Great docs @dcolligan, thanks for this. I've learned quite a bit from this! +1 (after adding in the requested example above)

**Comment by dcolligan** [comment]
@jeromekelleher updated

**Comment by shajoezhu** [comment]
:+1: Nice!

**Comment by jeromekelleher** [comment]
Git voodoo! +1.

**Pull request merged**